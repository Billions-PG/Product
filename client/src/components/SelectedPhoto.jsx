import React from 'react';
import PropTypes from 'prop-types';

const SelectedPhoto = ({ productImage}) => (
  <div>Selected photo</div>
);

export default SelectedPhoto;
