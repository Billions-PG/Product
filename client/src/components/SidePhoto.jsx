// todo - fix these eslint
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import React from 'react';
import PropTypes from 'prop-types';

const SidePhoto = ({ image, changeSelectedImg }) => (
  <img
    onClick={changeSelectedImg}
    className="side-photo"
    alt="randomly generated"
    src={image}
    data-testid="side-photo"
  />
);

SidePhoto.propTypes = {
  image: PropTypes.string.isRequired,
  changeSelectedImg: PropTypes.func.isRequired,
};

export default SidePhoto;
