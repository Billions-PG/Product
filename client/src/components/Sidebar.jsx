import React from 'react';
import PropTypes from 'prop-types';

const Sidebar = ({ seller, product }) => (
  <div id="product-sidebar">
    {/* Seller Name & Sales */}
    <span>{seller.name}</span>
    <div>
      <span>
        {seller.sales}
        {' '}
        sales
      </span>
      <span>&#9733;&#9733;&#9733;&#9733;&#9733;</span>
    </div>

    {/* Product Name, Price & Stock */}
    {
      seller.products !== undefined
      && (
      <div>
        <span>{product.name}</span>
        <div>
          <span>
            $
            {product.price}
          </span>
          {
            product.stock
              ? <span>&#10003; In stock</span>
              : <span>Out of stock</span>
          }
        </div>

        {/* Quantity &Sizes */}
        <div>
          <div>
            <span>Quantity</span>
            <select name="quantity" id="product-quantity">
              <option value="1">1</option>
              <option value="2">2</option>
              <option value="3">3</option>
              <option value="4">4</option>
              <option value="5">5</option>
            </select>
          </div>
        </div>
        {
          product.sizes
            ? (
              <div>
                <div>
                  <span>Quantity</span>
                  <select name="quantity" id="product-quantity">
                    <option value="1">1</option>
                    <option value="2">2</option>
                    <option value="3">3</option>
                    <option value="4">4</option>
                    <option value="5">5</option>
                  </select>
                </div>
              </div>
            )
            : <></>
        }

        {/* Buttons */}
        <div>
          <button type="button">Buy it now</button>
          <button type="button">Add to cart</button>
        </div>

        {/* Description */}
        <div>
          <span>Description</span>
          <span>{product.description}</span>
        </div>

      </div>
      )
    }
  </div>
);

Sidebar.propTypes = {
  seller: PropTypes.shape({
    name: PropTypes.string,
    sales: PropTypes.number,
    products: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      image: PropTypes.string,
      description: PropTypes.string,
      price: PropTypes.string,
      stock: PropTypes.number,
      sizes: PropTypes.bool,
      rating: PropTypes.number,
    })),
  }).isRequired,
  product: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    image: PropTypes.string,
    description: PropTypes.string,
    price: PropTypes.string,
    stock: PropTypes.number,
    sizes: PropTypes.bool,
    rating: PropTypes.number,
  }).isRequired,
};

export default Sidebar;
