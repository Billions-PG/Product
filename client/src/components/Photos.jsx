import React, { useEffect, useState } from 'react';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import { useParams } from 'react-router-dom';
import axios from 'axios';
import SelectedPhoto from './SelectedPhoto';
import SidePhoto from './SidePhoto';

//CWM adding for temp photo links
const DOSH_URL = "https://destinationcapstone.sfo2.digitaloceanspaces.com/";

const Photos = () => {
  const { prodId } = useParams();

  const [images, changeImages] = useState([]);
  const [selectedImg, changeSelectedImg] = useState('');

  const setState = async () => {
    const res = await axios.get(`http://3.141.24.146:3002/sellers/${prodId}`); // dev
    // const res = await axios.get(`http://54.90.53.234:3002/sellers/${prodId}`); // joey
    // const res = await axios.get(`http://3.15.40.71:3002/sellers/${prodId}`); // pablo
    // const res = await axios.get(`http://52.53.221.54:3002/sellers/${prodId}`); //jon
    console.log("PRINTING RESPONSE: ", res.data);

    //changeImages(res.data[0].products.filter((p) => p.id === Number(prodId))[0].images);
    //changeSelectedImg(res.data[0].products.filter((p) => p.id === Number(prodId))[0].images[0]);

    changeImages(res.data[0].images);
    changeSelectedImg(res.data[0].images[0]);
  };

  const changeSelected = (e) => {
    changeSelectedImg(e.target.src);
  };

  useEffect(() => {
    setState();
  }, [prodId]);

  return (
    <>
      <Container>
        <Row>
          <Col md={2}>
            {
              images.map((image) => (
                <SidePhoto
                  key={image}
                  changeSelectedImg={changeSelected}
                  image={image}
                  selectedImg={selectedImg}
                />
              ))
            }
          </Col>
          {/* </div> */}
          <Col>
            {
              selectedImg
              && (
                <SelectedPhoto selectedImg={selectedImg} />
              )
            }
          </Col>
        </Row>
      </Container>
    </>

  );
};

export default Photos;
