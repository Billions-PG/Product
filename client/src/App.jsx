import React from 'react';
import ReactDom from 'react-dom';
import axios from 'axios';
import Sidebar from './components/Sidebar';
import Photos from './components/Photos';

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      seller: {},
      product: {},
      selectedImg: '',
    };

    this.changeSelectedImg = this.changeSelectedImg.bind(this);
  }

  componentDidMount() {
    this.getRandomSellerAndProduct();
    setTimeout(() => {
      const { product } = this.state;
      this.setState({ selectedImg: product.images[0] });
    }, 100);
  }

  getRandomSellerAndProduct() {
    const id = Math.floor(Math.random() * Math.floor(100)) + 1;
    axios.get(`/sellers/${id}`)
      // set seller
      .then((res) => {
        this.setState({ seller: res.data });
        return res.data.products;
      })
      // set product
      .then((products) => {
        products.forEach((product) => {
          if (product.id === id) {
            this.setState({ product });
          }
        });
      })
      .catch((err) => err);
  }

  changeSelectedImg(e) {
    this.setState({ selectedImg: e.target.src });
  }

  render() {
    const { seller, product, selectedImg } = this.state;
    return (
      <div id="product-app">
        {
          product.images
          && (
            <Photos
              changeSelectedImg={this.changeSelectedImg}
              images={product.images}
              selectedImg={selectedImg}
            />
          )
        }
        <Sidebar product={product} seller={seller} />
      </div>
    );
  }
}

export default App;

ReactDom.render(<App />, document.getElementById('app'));
