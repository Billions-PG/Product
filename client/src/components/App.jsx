import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import {
  HashRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';
import Sidebar from './Sidebar';
import Photos from './Photos';
import styles from '../styles.module.css';

const App = () => (
  <Router>
    <Switch>
      <Route path="/product/:prodId">
        <Container className={`${styles.app}`}>
          <Row className="d-flex justify-content-center">
            <Col md={8}>
              <Photos />
            </Col>
            <Col fluid="true">
              <Sidebar />
            </Col>
          </Row>
        </Container>
      </Route>
    </Switch>
  </Router>
);

export default App;
