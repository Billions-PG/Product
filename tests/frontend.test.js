/* eslint-disable react/jsx-filename-extension */
/**
* @test-environment jsdom
*/
import 'jsdom-global/register';
import '@testing-library/jest-dom';
import React from 'react';
import { render } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';

import Photos from '../client/src/components/Photos';
import SelectedPhoto from '../client/src/components/SelectedPhoto';
import Sidebar from '../client/src/components/Sidebar';
import SidePhoto from '../client/src/components/SidePhoto';
import App from '../client/src/components/App';

// product = sizes: false, stock: 0
// productAlt = sizes: true, stock: >0
const { product, seller, productAlt } = require('./mockData');

describe('<App />', () => {
  it('Renders the App', () => {
    render(<App />);
  });
});

// Photos
describe('<Photos />', () => {
  const mock = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  const renderPhotos = () => render(<Photos
    changeSelectedImg={mock}
    images={product.images}
    selectedImg={product.images[0]}
  />);

  it('Renders the Photos component', () => {
    renderPhotos();
  });
});

// SidePhoto
describe('<SidePhoto />', () => {
  let mock;
  beforeAll(() => {
    mock = jest.fn();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  const renderSidePhoto = () => render(<SidePhoto
    changeSelectedImg={mock}
    image={product.images[0]}
  />);

  it('Renders the SidePhoto component', () => {
    renderSidePhoto();
  });

  // it('Calls function with target', () => {
  //   const { getAllByTestId } = renderSidePhoto();
  //   const images = getAllByTestId('side-photo');
  //   userEvent.click(images);
  // });
});

// SelectedPhoto
describe('<SelectedPhoto />', () => {
  it('Renders the SelectedPhoto component', () => {
    render(<SelectedPhoto
      selectedImg={product.images[0]}
    />);
  });
});

// Sidebar
describe('<Sidebar />', () => {
  const renderSidebar = (s = seller, p = product) => render(<Sidebar
    seller={s}
    product={p}
  />);

  it('Renders the Sidebar component', () => {
    renderSidebar();
  });

  it('Renders span with \'out of stock\' message at zero stock', () => {
    const { getByTestId } = renderSidebar(undefined, productAlt);
    const stock = getByTestId('stock');
    expect(stock).toContainHTML('<span data-testid="stock">Out of stock</span>');
  });

  it('Renders span with \'In stock\' message at zero stock', () => {
    const { getByTestId } = renderSidebar(undefined, product);
    const stock = getByTestId('stock');
    expect(stock).toContainHTML('<span data-testid="stock">âœ“ In stock</span>');
  });

  it('Renders size choices when applicable', () => {
    const { getByTestId } = renderSidebar(undefined, productAlt);
    const sizes = getByTestId('size');
    expect(sizes).toMatchSnapshot();
  });
});
