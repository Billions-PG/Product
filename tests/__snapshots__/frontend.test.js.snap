// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> Renders size choices when applicable 1`] = `
<select
  data-testid="size"
  id="product-quantity"
  name="quantity"
>
  <option
    value="XS"
  >
    XS
  </option>
  <option
    value="S"
  >
    S
  </option>
  <option
    value="M"
  >
    M
  </option>
  <option
    value="L"
  >
    L
  </option>
  <option
    value="XL"
  >
    XL
  </option>
</select>
`;
